crates/bevy_app/src/plugin_group.rs

```diff
@@ -2,6 +2,95 @@ use crate::{App, AppError, Plugin};
 use bevy_utils::{tracing::debug, tracing::warn, HashMap};
 use std::any::TypeId;
 
+/// A macro for generating a well-documented [`PluginGroup`] from a list of [`Plugin`] paths.
+///
+/// Every plugin must implement the [`Default`] trait.
+///
+/// # Example
+///
+/// ```
+/// # use bevy_app::*;
+/// # mod velocity {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct VelocityPlugin;
+/// #   impl Plugin for VelocityPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod collision { pub mod capsule {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct CapsuleCollisionPlugin;
+/// #   impl Plugin for CapsuleCollisionPlugin { fn build(&self, _: &mut App) {} }
+/// # } }
+/// # #[derive(Default)] pub struct TickratePlugin;
+/// # impl Plugin for TickratePlugin { fn build(&self, _: &mut App) {} }
+/// # mod features {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct ForcePlugin;
+/// #   impl Plugin for ForcePlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// # mod web {
+/// #   use bevy_app::*;
+/// #   #[derive(Default)] pub struct WebCompatibilityPlugin;
+/// #   impl Plugin for WebCompatibilityPlugin { fn build(&self, _: &mut App) {} }
+/// # }
+/// plugin_group!(
+///     PhysicsPlugins {
+///         // Due to local ambiguity issues, you have to
+///         // use a colon before the plugin's name.
+///         :TickratePlugin,
+///         // Due to local ambiguity issues, you have to
+///         // use 3 colons for the last part of the path.
+///         collision::capsule:::CapsuleCollisionPlugin,
+///         velocity:::VelocityPlugin,
+///         #[cfg(feature = "external_forces")]
+///         features:::ForcePlugin,
+///         // You can add any attribute you want like this, but it won't be documented.
+///         #[custom(cfg(target_arch = "wasm32"))]
+///         web:::WebCompatibilityPlugin
+///     }
+/// );
+/// ```
+#[macro_export]
+macro_rules! plugin_group {
+    (
+        $(#[$group_meta:meta])*
+        $group:ident {
+            $(
+                $(#[cfg(feature = $feature:literal)])?
+                $(#[custom($plugin_meta:meta)])*
+                $($plugin_path:ident::)* : $plugin_name:ident
```

Comment:
I don't think an `ident` works well here. Perhaps it could be a:

- `path`: _TypePath_
- `ty`: _Type_
  - I don't know the difference between a type path and just a type, but it may be worth investigating.
- `expr`: _Expression_
  - Instead of requiring every plugin implement default, you could instead require an expression and manually write `MyPlugin::default()` for each.

Reply:
The reason it looks like this is because I need to get 2 things from the path (for documentation): The short name, and the full path. The full path part is easy, but getting the short name is not. The way I solved this is by seperating the short name from the rest of the path with an additional colon (This is needed due to local ambiguity or whatever).

Reply:
That makes sense, though I still think the triple colon may be confusing. Perhaps you could phrase it like a match statement?

```rust
plugin_group! {
    MyPlugins {
        bevy_app => AppPlugin,
        crate => MyCoolPlugin,
        self => AnotherPlugin,
    }
}
```

I personally think it looks better, but it's just a suggestion. What you have now works fine too. :)

Reply:
I'm not really a fan of that syntax, since it doesn't really look like a path.

Reply:
Ok, that's fine. :)